src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts

```diff
@@ -522,4 +534,22 @@ export class ProgrammingExerciseConfigureGradingComponent implements OnInit, OnD
             )
             .subscribe();
     }
+
+    public isSumOfWeightsGreaterThanZero(testCaseUpdates: ProgrammingExerciseTestCaseUpdate[]): boolean {
+        // Copy the existing test cases and update the test cases that were changed
+        const copyTestCases = cloneDeep(this.testCases);
+        testCaseUpdates.forEach((testCaseUpdate) => {
+            const index = copyTestCases.findIndex((testCase) => testCase.id === testCaseUpdate.id);
+            if (index !== -1) {
+                copyTestCases[index] = testCaseUpdate;
+            }
+        });
+        // Make sure that the at least one test weight is greater than 0, so that students can still reach a score of 100%
+        const sumOfWeightNewTestCases = copyTestCases.reduce((sum, testCase) => sum + (testCase.weight ?? 0), 0);
+        if (sumOfWeightNewTestCases <= 0) {
+            return false;
+        } else {
+            return true;
+        }
```

Comment:
This can/should simplify to `return sumOfWeightNewTestCases > 0;`.

Reply:
Done in `16df0615c`